source/Octopus.Tentacle/Configuration/Proxy/ProxyPasswordMaskValuesProvider.cs

```diff
@@ -9,15 +9,15 @@ namespace Octopus.Tentacle.Configuration.Proxy
     
     public interface IProxyPasswordMaskValuesProvider
     {
-        IEnumerable<string> GetProxyPasswordMaskValues(string proxyPassword);
+        IEnumerable<string> GetProxyPasswordMaskValues(string? proxyPassword);
     }
     
     public class ProxyPasswordMaskValuesProvider : IProxyPasswordMaskValuesProvider
     {
         static readonly Regex UrlEncodedCharactersRegex = new Regex(@"%[A-F0-9]{2}", RegexOptions.Compiled);
-        public IEnumerable<string> GetProxyPasswordMaskValues(string proxyPassword)
+        public IEnumerable<string> GetProxyPasswordMaskValues(string? proxyPassword)
         {
-            if (string.IsNullOrEmpty(proxyPassword)) 
+            if (proxyPassword == null || string.IsNullOrEmpty(proxyPassword)) 
```

Comment:
Looks like the existing condition should already check for nullability?

Reply:
It doesn't and results in a compiler error.

Reply:
damn